### DISCUSSION TITLE
Are there anyway I can control 3x 1 channel lights with just one ESP?

### DISCUSSION THREAD

**Discussion by beobuild** [arendst/Tasmota#21772] [Q&A] [answered]
Hi.

I'm looking for a way to control my 3 lights by using only one ESP.

I've already tried to set the PWM Pins, but they result in a 2 channels light setup.

I'm using the latest build of Tasmota with Matter and Google Home.

Any idea or suggestions?

**Comment by Jason2866** [comment, marked as answer]
Short answer, no. Long answer you have to code yourself. Probably the easiest way with Berry. NOT saying it is easy and not complex. There is no plan to enhance the light driver in Tasmota to support more lights, since it would need major and many changes in code. There is a low chance that a PR for this will be merged.

You can control GPIOS separate as PWM channels, loosing all light features.

**Reply by sfromis** [reply]
Using `SetOption68` (`MultiPWM`) you get to having the PWM channels with separate web UI buttons/sliders pretty much acting similar to lights without doing any coding, but as there are different commands to control them, it is not quite the same, or "full features". Still, this should work out for simple usage.

**Reply by beobuild** [reply]
> Using `SetOption68` (`MultiPWM`) you get to having the PWM channels with separate web UI buttons/sliders pretty much acting similar to lights without doing any coding, but as there are different commands to control them, it is not quite the same, or "full features". Still, this should work out for simple usage.

Appreciate for your help!
Hmm, just wondering, how would that be registered on Matter?

Edit: It did it! Thanks for your help :)

**Reply by sfromis** [reply]
It not being "full features" includes that it AFAIK would not be registered on Matter. Still, I guess it could be possible to do something with:
https://tasmota.github.io/docs/Matter/#matter-virtual-devices
I'd opt for doing the integration via Berry in such a case.

**Reply by beobuild** [reply]
I've already built up a rule like so, but seems like I can't change the brightness

```
Rule1
on mtrreceived#Light1#power==1 do Power1 1 endon
on mtrreceived#Light1#power==0 do Power1 0 endon
on mtrreceived#Light1#Dimmer do Channel1 mtrreceived#Light1#Dimmer endon
on Power1#State==1 do mtrupdate {"name":"Light1","power":1} endon 
on Power1#State==0 do mtrupdate {"name":"Light1","power":0} endon
```

Can anyone explain this for me? I'm fairly new to Tasmota. :/

**Reply by sfromis** [reply]
I'm assuming that your dimmer rule above did not work, it looks too odd, but **if** you see a Dimmer value in a `MtrReceived` payload, to match the trigger you use, you should use `%value%` after the `do`, to pick up the value.

Additionally, while I can't explain details, I do observe that the docs for the `MtrUpdate` command mentions attribues like `Bri` for brightness, along the `Power` attribute you're already using.
https://tasmota.github.io/docs/Matter/#mtrupdate-command

**Reply by beobuild** [reply]
Hi, just a quick followup here,

I've replaced `mtrreceived#Light1#Dimmer` with `%value%` after the `do`.

It's working good rn, and I'm looking into the `Bri` attribute for brightness.

I'm still not pretty sure how to get it going rn, here's my `MtrReceived` data:

```
{"MtrReceived":{"Light1":{"Ep":10,"Name":"Light1","Bri":112,"Dimmer":44,"Power":1}}}
```

I'm still not sure how the `Bri` value is defined, can you clear that up for me?

Btw, the `Dimmer` actually equals to the brightness value shown in Google Home.

Thanks ;)

**Reply by sfromis** [reply]
I don't know much about details, but comments in the Tasmota source code suggests that `Bri` *may* be related to brightness without gamma correction.

**Reply by beobuild** [reply]
Well, it still works without updating the `Bri` variable back to Matter.

Thanks for your help ;)

**Comment by beobuild** [comment]
Here's a workaround that you can use (creds to @sfromis)

First off, enable `MultiPWM`: `SetOption68 1`

Setup all modules in Configuration, then head to the Matter configuration, create your virtual devices:
![image](https://github.com/user-attachments/assets/c0948637-6369-4cff-b490-dc11e659a757)

Create a new rule (substitute `Light1` and `Power1` with your config):

```
Rule1
on mtrreceived#Light1#power==1 do Power1 1 endon
on mtrreceived#Light1#power==0 do Power1 0 endon
on mtrreceived#Light1#Dimmer do Channel1 %value% endon
on Channel1 do mtrupdate {"name":"Light1","Bri":%value%} endon
on Power1#State==1 do mtrupdate {"name":"Light1","power":1} endon 
on Power1#State==0 do mtrupdate {"name":"Light1","power":0} endon
``` 

That's it ;)
